packages/kbn-test/src/auth/session_manager.ts

```diff
@@ -54,7 +54,12 @@ export class SamlSessionManager {
     this.isCloud = options.isCloud;
     this.log = options.log;
     // if the rolesFilename is provided, respect it. Otherwise use DEFAULT_ROLES_FILE_NAME.
-    const rolesFile = rolesFilename ? rolesFilename : this.DEFAULT_ROLES_FILE_NAME;
+    let rolesFile: string;
+    if (process.env.ROLES_FILENAME_OVERRIDE) {
+      rolesFile = process.env.ROLES_FILENAME_OVERRIDE;
+    } else {
+      rolesFile = rolesFilename ? rolesFilename : this.DEFAULT_ROLES_FILE_NAME;
+    }
```

Comment:
I'm wondering why another override mechanism is needed. You can already override the default roles file by passing an alternative file to the `SamlSessionManager` constructor. And I see this is already done here in the context of proxy organisations for the security solution: https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/management/cypress/support/saml_authentication.ts#L36-L45.
Not sure if this can be reused for what you have in mind, but worth to check out and maybe avoid code duplication?

Reply:
Synced with @dmlemeshko (thanks for your help) and fixed this. Thanks a lot for pinging on this.